### PR TITLE
Emailable report

### PR DESCRIPTION
**Pull request by jessicamarshall**

### CODE REVIEW

File: src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+		String reportTemplate = initReportTemplate();
+		String htmlString;
+		
+		getTestSuiteDetails(suites);
+		
+		htmlString = suiteSummaryRows.stream().collect(Collectors.joining());
+		reportTemplate = reportTemplate.replaceFirst("<tbody id=\"suiteSummaryTable\">", String.format("%s</tbody>", htmlString));
+		htmlString = suiteDetailRows.stream().collect(Collectors.joining());
+		reportTemplate = reportTemplate.replaceFirst("<tbody id=\"suiteDetailTable\">", String.format("%s</tbody>", htmlString));
+
```
Comment (JustinPhlegar) [inline src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java]: Is this duplicated code?

Comment (JustinPhlegar) [reply src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java]: Disregard, I see now one is Summary table and other is Detail table, Can they be spaced and comments added so we don't accidentally think it is dupe code and remove in future?

File: src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+		new File(outputDirectory).mkdirs();
+		try {
+			PrintWriter reportWriter = new PrintWriter(new BufferedWriter(new FileWriter(new File(outputDirectory, "automation-email-report.html"))));
+			reportWriter.println(reportTemplate);
+			reportWriter.flush();
+			reportWriter.close();
+		} catch (IOException e) {
+			throw new AutomationException("Could not write to report template.  Error: " + e.getMessage());
+		}
+	}
```
Comment (JustinPhlegar) [inline src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java]: If reportWriter.println fails, the stream is never closed. Recommend the following code.
```
        try (PrintWriter reportWriter = new PrintWriter(new BufferedWriter(new FileWriter(new File(outputDirectory, "automation-email-report.html"))), true)){            
            reportWriter.println(reportTemplate);
        } catch (IOException e) {
            throw new AutomationException("Could not write to report template.  Error: " + e.getMessage());
        }
```

- try-with-resource methodology exposes stream to both try block and catch block if needed. 
- PrintWriter also inherits auto-closable. So when PrintWriter is created in try-with-resources, it automatically closes the stream as apart of cleanup like a you would do in a finally block.
- Adding the true boolean in PrintWriter constructor states that after a println, the stream will flush its buffer automatically

File: src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java (hunk 1 above)
Comment (jessicamarshall) [reply src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java]: Sure thing - I forgot to add comments in this method, so went ahead and added them.

File: src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java (hunk 2 above)
Comment (jessicamarshall) [reply src/main/java/com/orasi/utils/listeners/CustomizedEmailableReport.java]: Good catch - you had mentioned previously about try with resources so is a good example of usage.  Will update code.

**Pull request merged**